src/Corax/Indexing/IndexWriter.cs

```diff
@@ -1487,8 +1487,9 @@ private void RecordTermsForEntries(in NativeList<TermInEntryModification> entrie
                         recordedTermContainerId |= 1; // marker!
                         recordedTerm.Long = entries.Long.Value;
 
-                        // only if the double value can not be computed by casting from long, we store it 
-                        if (entries.Double != null && entries.Double.Value.AlmostEquals(recordedTerm.Long) == false)
+                        // only if the double value can not be computed by casting from long, we store it
+                        // Since we store double values internally as longs, converted via BitConverter, it is good to check whether equal elements have exactly the same value in this form.
+                        if (entries.Double != null && BitConverter.DoubleToInt64Bits(entries.Double.Value) != BitConverter.DoubleToInt64Bits(recordedTerm.Long))
                         {
```

Comment:
That's a reasonable approach. We ensure at the source that we are encoding the proper thing even accounting for the issues at https://github.com/dotnet/runtime/issues/96416